recipes/jasper/all/conanfile.py

```diff
@@ -25,88 +28,89 @@ class JasperConan(ConanFile):
         "with_libjpeg": "libjpeg",
     }
 
-    generators = "cmake", "cmake_find_package"
-    _cmake = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
     @property
-    def _build_subfolder(self):
-        return "build_subfolder"
+    def _cmake_gen(self):
+        self.conf.get("tools.cmake.cmaketoolchain:generator", check_type=str, default="Ninja")
 
     def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.cppstd
-        del self.settings.compiler.libcxx
+            try:
+                del self.options.fPIC
+            except:
+                pass
+        try:
+            del self.settings.compiler.cppstd
+        except Exception:
+            pass
+        try:
+            del self.settings.compiler.libcxx
+        except Exception:
+            pass
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
         if self.options.with_libjpeg == "libjpeg-turbo":
-            self.requires("libjpeg-turbo/2.1.2")
+            self.requires("libjpeg-turbo/2.1.4")
         elif self.options.with_libjpeg == "libjpeg":
-            self.requires("libjpeg/9d")
+            self.requires("libjpeg/9e")
+
+    def build_requirements(self):
+        self.tool_requires("ninja/1.11.0")
```

Comment:
that's quite a strong requirement, what's wrong here?

Reply:
Good question, it's because of this issue: https://github.com/conan-io/conan-center-index/issues/13159

Using Ninja we can skip that error, but it's clear that this workaround is not what we want, but a new version of the SDK.